hnn_core/network.py

```diff
@@ -869,20 +869,33 @@ def _instantiate_drives(self, n_trials=1):
             for drive in self.external_drives.values():
                 event_times = list()  # new list for each trial and drive
 
-                # loop over drive 'cells' and create event times for each
+                # loop over drives (one for each target cell population)
+                # and create event times
                 for this_cell_drive_conn in drive['conn'].values():
-                    for drive_cell_gid in this_cell_drive_conn['src_gids']:
-                        event_times.append(_drive_cell_event_times(
+                    if drive['cell_specific']:
+                        for drive_cell_gid in this_cell_drive_conn['src_gids']:
+                            event_times.append(_drive_cell_event_times(
+                                drive['type'], this_cell_drive_conn,
+                                drive['dynamics'], trial_idx=trial_idx,
+                                drive_cell_gid=drive_cell_gid,
+                                seedcore=drive['seedcore'])
+                            )
+                    else:
+                        # cell_specific=False should only have one src_gid
+                        assert len(this_cell_drive_conn['src_gids']) == 1
+
```

Comment:
```suggestion
# Only return empty event times if all cells have no events
```

Reply:
I get the feeling we may need a "here there be dragons" message of our own as this code is dealing with edge cases that are not immediately obvious (empty events due to 0 weight, or the parameters of the drive)

Reply:
done, added this comment. By the way, I was surprised that this part of the code was not tested directly. Moving forward, we should add tests that check these corner cases